x-pack/platform/plugins/shared/index_management/public/application/sections/home/index_list/index_table/get_doc_count.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Subject, bufferTime, concatMap, filter, share, scan } from 'rxjs';
+import type { HttpSetup } from '@kbn/core-http-browser';
+import { INTERNAL_API_BASE_PATH } from '../../../../../../common/constants';
+
+export const docCountApi = (httpSetup: HttpSetup) => {
+  const subj = new Subject<string>();
+  const abortController = new AbortController();
+
+  const observable = subj.pipe(
+    // Buffer indices for 100ms before making a request
+    bufferTime(100),
+    filter((indices) => indices.length > 0),
+    concatMap((indices) =>
+      httpSetup.post<Record<string, number>>(`${INTERNAL_API_BASE_PATH}/index_doc_count`, {
+        body: JSON.stringify({ indexNames: indices }),
+        signal: abortController.signal,
+      })
+    ),
+    scan((acc, response) => ({ ...acc, ...response }), {} as Record<string, number>),
+    share()
```

Comment:
Right now each emission is just the latest batch response map, and `DocCountCell` treats “missing key” as “still loading”. That creates a real UX bug when the visible set expands (e.g. toggling include hidden): the new batch can emit a map that doesn’t contain previously visible indices, so those already-mounted rows flip to spinner and never re-request.

Suggested fix: merge responses into an accumulated map before sharing/replaying, e.g.

- `scan((acc, batch) => ({ ...acc, ...batch }), {})`
- and ideally `shareReplay({ bufferSize: 1, refCount: true })` (so all cells see the latest accumulated state)

This isn’t “cache for speed”; it keeps the stream’s emitted state consistent so previously resolved counts don’t disappear when new batches arrive.

Reply:
>That creates a real UX bug when the visible set expands (e.g. toggling include hidden): the new batch can emit a map that doesn’t contain previously visible indices, so those already-mounted rows flip to spinner and never re-request.

I'm still not understanding the flow here although its no longer important but I'm still curious 

>so those already-mounted rows flip to spinner and never re-request.

Why don't they rerequest? I don't understand the state where they lose state and don't rerequest.

---

looks like I beat you to the punch on using scan, will look at shareReplay